tests/platform_helper/domain/test_pipelines.py

```diff
@@ -122,6 +122,29 @@ def test_generate_pipeline_command_generate_terraform_files_for_environment_pipe
     )
 
 
+@freeze_time("2024-10-28 12:00:00")
+@patch("dbt_platform_helper.jinja2_tags.version", new=Mock(return_value="v0.1-TEST"))
+def test_generate_pipeline_generates_expected_terraform_manifest_when_no_deploy_repository_key(
+    fakefs,
+    platform_config_for_env_pipelines,
+):
+
+    app_name = "test-app"
+    # deploy_repository key set on test_fixture so remove it
+    platform_config_for_env_pipelines.pop("deploy_repository")
```

Comment:
Can we use `delte` as that is what we are doing really?
```suggestion
    platform_config_for_env_pipelines.delete("deploy_repository")
```

Reply:
.delete() isn't a function in Python,

Other option is `del platform_config_for_env_pipelines['deploy_repository']`

but this isn't as safe as a `.pop()` operation

Reply:
You are correct. My IDE suggests that it is, but it's wrong, and I didn't run the tests after I typed it.

Reply:
Why is `some_dict.pop("thing")` safer than `del some_dict["thing"]`?

Reply:
Not relevant to this example but `.pop()` is generally preferred since you have the option to make the dict not throw a key error when the key isn't present, i.e.

`my_dict.pop('key1', None)`

Reply:
Maybe there is an argument for del in this case if we want to be *pedantic*, either way both methods achieve the same thing as far as this test is concerned ¯\_(ツ)_/¯